### PR TITLE
Updating Azure Search's template to 2015-08-19

### PR DESCRIPTION
**Pull request by seansaleh**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
* Changed template to use 2015-08-19
* Added support for hostingMode
* Added specific options for SKU

### Description of the change
Updating Azure Search's template to 2015-08-19

FYI @brjohnstmsft

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@seansaleh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (seasa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: 101-azure-search-create/azuredeploy.json [hunk 1]
```diff
@@ -40,20 +47,32 @@
...
+    },
+    "hostingMode": {
+      "type": "string",
+      "defaultValue": "default",
+      "allowedValues": [
+        "default",
+        "highDensity"
+      ],
+      "metadata": {
+        "description": "Applicable only for SKU set to standard3. You can set this property to enable a single, high density partition that allows up to 1,000 indexes, which is much higher than the maximum indexes allowed for any other SKU."
```
Comment (brjohnstmsft) [inline 101-azure-search-create/azuredeploy.json]: nit: Leave the comma out of 1000 as this is locale-dependent.

**Comment by brjohnstmsft** [comment]
👍 LGTM

File: 101-azure-search-create/azuredeploy.json (hunk 1 above)
Comment (seansaleh) [inline 101-azure-search-create/azuredeploy.json]: Good point, will do. I pulled this from our docs.

**Pull request merged**